sale_triple_discount/models/sale_order_line.py

```diff
@@ -0,0 +1,147 @@
+    # the main discount is 24.7885 % (and not 24.79)
+    discount = fields.Float(
+        string="Total Disc (%)",
+        store=True,
+        compute="_compute_discount",
+        compute_sudo=True,
+        precompute=True,
+    )
+    discount1 = fields.Float(
+        string="Disc. 1 (%)",
+        digits="Discount",
+        compute="_compute_discount1",
+        store=True,
+        compute_sudo=True,
+        precompute=True,
+        readonly=False,
+    )
+    discounting_type = fields.Selection(
+        selection=[("additive", "Additive"), ("multiplicative", "Multiplicative")],
+        default="multiplicative",
+        required=True,
+        help="Specifies whether discounts should be additive "
+        "or multiplicative.\nAdditive discounts are summed first and "
+        "then applied.\nMultiplicative discounts are applied sequentially.\n"
+        "Multiplicative discounts are default",
+    )
+
+    @api.constrains("discounting_type")
+    def _discounting_type_additive_not_allowed(self):
+        # FIXME: see https://github.com/OCA/sale-workflow/issues/3649
+        if any(rec.discounting_type == "additive" for rec in self):
+            raise ValidationError(
+                _(
+                    "Additive discount type is not fully implemented."
+                    " See https://github.com/OCA/sale-workflow/issues/3649 "
+                )
+            )
+
+    def _get_final_discount(self):
+        self.ensure_one()
+        if self.discounting_type == "additive":
+            return self._additive_discount()
+        elif self.discounting_type == "multiplicative":
+            return self._multiplicative_discount()
+        else:
+            raise ValidationError(
+                _("Sale order line %(name)s has unknown discounting type %(dic_type)s")
+                % {"name": self.name, "disc_type": self.discounting_type}
+            )
+
+    def _additive_discount(self):
+        self.ensure_one()
+        discount = sum(
+            self[x] or 0.0 for x in self._get_multiple_discount_field_names()
+        )
+        if discount <= 0:
+            return 0
+        elif discount >= 100:
+            return 100
+        return discount
+
+    def _multiplicative_discount(self):
+        self.ensure_one()
+        return self._get_aggregated_multiple_discounts(
+            [self[x] for x in self._get_multiple_discount_field_names()]
+        )
+
+    @api.depends(
+        lambda self: self._get_multiple_discount_field_names()
+        + ["product_id", "product_uom", "product_uom_qty"]
+    )
+    def _compute_discount(self):
+        # Base Odoo just continues instead of assigning to 0 in this case
+        # but we depend on the super() value resetting to a discount unpolluted
+        # by the extra fields before taking them into account
+        for line in self:
+            if not (line.pricelist_item_id and line.pricelist_item_id._show_discount()):
```

Comment:
```suggestion
        discount_enabled = self.env['product.pricelist.item']._is_discount_feature_enabled()
        for line in self:
            if not (line.order_id.pricelist_id and discount_enabled):
```
Base 18.0 `_compute_discount` looks a bit different to 17.0
https://github.com/odoo/odoo/blob/322c6d0468bf79e9d29e1375c49aa13d4a7b7a67/addons/sale/models/sale_order_line.py#L758-L777

Reply:
@sbiosca-s73 I'd say this change is necessary to avoid future headaches of possible edge cases where the 17.0 `if` passes and the `18.0` one doesn't or vice versa